File: app/src/main/java/com/bw/jinrong/fragment/InvestFragment.java
package com.bw.jinrong.fragment;


import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;
import android.support.v4.view.ViewPager;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import android.widget.ImageView;
import android.widget.TextView;
import com.bw.base.BaseFragment;
import com.bw.jinrong.R;
import com.bw.view.TabPageIndicator;

import java.util.ArrayList;
import java.util.List;

/**
 * A simple {@link Fragment} subclass.
 */
public class InvestFragment extends BaseFragment {

    private View mView;

    @Override
    protected void initData() {
        initFragments();

        MyAdapter adapter = new MyAdapter(getFragmentManager());
//        mView = getBaseView();
        ViewPager vp_invest = mView.findViewById(R.id.vp_invest);
        vp_invest.setAdapter(adapter);

        TabPageIndicator tabpage_invest = mView.findViewById(R.id.tabpage_invest);
        tabpage_invest.setViewPage(vp_invest);

        initTitle();

    }

    private List<Fragment> fragmentList = new ArrayList<>();
    private void initFragments() {
        ProductListFragment productListFragment = new ProductListFragment();
        ProductRecommondFragment productRecommondFragment = new ProductRecommondFragment();
        ProductHotFragment productHotFragment = new ProductHotFragment();

        fragmentList.add(productListFragment);
        fragmentList.add(productRecommondFragment);
        fragmentList.add(productHotFragment);

    }

    protected void initTitle(){
        ImageView iv_title_back = mView.findViewById(R.id.iv_title_back);
        TextView tv_title = mView.findViewById(R.id.tv_title);
        ImageView iv_title_setting = mView.findViewById(R.id.iv_title_setting);

        iv_title_back.setVisibility(View.GONE);
        tv_title.setText("投资");
        iv_title_setting.setVisibility(View.GONE);

    }

    @Override
    protected int setView() {
        return R.layout.fragment_invest;
    }

    @Override
    public void onConnected() {

    }

    @Override
    public void onDisConnected() {

    }

    private class MyAdapter extends FragmentPagerAdapter {

        public MyAdapter(FragmentManager fm) {
            super(fm);
        }

        @Override
        public Fragment getItem(int i) {
            return fragmentList.get(i);
        }

        @Override
        public int getCount() {
            return 0;
        }
    }
}

File: app/src/main/java/com/bw/jinrong/fragment/InvestFragment.kt
package com.bw.jinrong.fragment


import android.support.design.widget.TabLayout
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter
import android.support.v4.view.ViewPager
import android.view.View

import android.widget.ImageView
import android.widget.TextView
import com.bw.base.BaseFragment
import com.bw.common.AppNetConfig
import com.bw.common.UIUtils
import com.bw.jinrong.R
import com.bw.jinrong.bean.InvestBean
import com.bw.jinrong.presenter.HomePresenter

import java.util.ArrayList

/**
 * A simple [Fragment] subclass.
 */
class InvestFragment : BaseFragment() {

    var mView: View? = null

    var fragmentList:ArrayList<Fragment> = ArrayList()

    override fun initData() {
        initFragments()

        mView = baseView

        var adapter = MyAdapter(childFragmentManager,fragmentList)
        //mView = getBaseView();
        val vp_invest = mView?.findViewById<ViewPager>(R.id.vp_invest)
        vp_invest?.adapter = adapter

        val tabpage_invest = mView?.findViewById<TabLayout>(R.id.tabpage_invest)
        tabpage_invest?.setupWithViewPager(vp_invest)

        initTitle()

        val homePresenter = HomePresenter(AppNetConfig.PRODUCT, InvestBean::class.java)
//        homePresenter.attachView(this)
        homePresenter.doHttpRequest()

    }

    private fun initFragments() {
        val productListFragment = ProductListFragment()
        val productRecommondFragment = ProductRecommondFragment()
        val productHotFragment = ProductHotFragment()

        fragmentList.add(productListFragment)
        fragmentList.add(productRecommondFragment)
        fragmentList.add(productHotFragment)

    }

    private fun initTitle() {
        var iv_title_back = mView?.findViewById<ImageView>(R.id.iv_title_back)
        var tv_title = mView?.findViewById<TextView>(R.id.tv_title)
        var iv_title_setting = mView?.findViewById<ImageView>(R.id.iv_title_setting)

        iv_title_back?.visibility = View.GONE
        tv_title?.text = "投资"
        iv_title_setting?.visibility = View.GONE

    }

    override fun setView(): Int {
        return R.layout.fragment_invest
    }

    override fun onConnected() {

    }

    override fun onDisConnected() {

    }

    class MyAdapter(fm: FragmentManager,fragmentList:MutableList<Fragment>) : FragmentPagerAdapter(fm) {

        var fragmentList:MutableList<Fragment> = fragmentList

        override fun getItem(i: Int): Fragment {
            return fragmentList[i]
        }

        override fun getCount(): Int {
            return if (fragmentList == null){
                0
            }else {
                fragmentList.size
            }
        }

        override fun getPageTitle(position: Int): CharSequence? {
            return UIUtils.getStringArr(R.array.invest_tab)[position]
        }

    }
}
